content/docs/command-reference/commit.md

```diff
@@ -17,65 +17,52 @@ positional arguments:
-iterations on an experiment). The `--no-commit` option prevents it (on the
-commands where it's available). The file hash is still computed and added to the
-`dvc.lock` or `.dvc` file, but the actual data is not cached. And this is where
-the `dvc commit` command comes into play: It performs that last step when
-needed.
-
-Note that it's best to avoid the last three scenarios. They essentially
-force-update the `dvc.lock` or `.dvc` files and save data to cache. They are
-still useful, but keep in mind that DVC can't guarantee reproducibility in those
-cases.
+This command can be useful for several scenarios, when
+[stages](/doc/command-reference/run) are under development, or to force-update
+the cache without having to execute stage commands again (avoiding `dvc repro`).
+
+> Note that this does not register project versions. That can be done with
+> `git commit` (if using Git).
+
+Normally, `dvc repro` and `dvc run` finish up with the same steps as `dvc add`,
+for each <abbr>output</abbr> involved. In summary:
+
+- Compute the hash value of the file or directory and save it in the `dvc.lock`
+  or `.dvc` file.
+- If using Git, append the file/directory path to `.gitignore`.
+- Store the data in question in the <abbr>cache</abbr>.
+
+For stages under development, the last step can be skipped with the
+`--no-commit` option of those commands. This avoids caching unfinished data
+(hash values are still calculated and added to `dvc.lock` or `.dvc` files). This
+is where `dvc commit` comes into play: It performs that last step when needed.
+
+💡 For convenience, a pre-commit Git hook is available to remind you to
+`dvc commit` when needed. See `dvc install` for more info.
+
+Other scenarios include:
+
+- Often we edit source code, configuration, or input data in a way that doesn't
+  cause changes to any outputs, for example reformatting data, adding code
+  comments, etc. However, DVC notices all changes to <abbr>dependencies</abbr>
+  and expects you to reproduce the corresponding stages. Use `dvc commit`
+  instead to force accepting the changes without having to `dvc repro`.
+
+- Sometimes, after executing a stage, we realize that we forgot to specify some
+  of its dependencies or outputs in `dvc.yaml`. Fortunately it's possible to
+  [add the missing deps/outs](/docs/user-guide/how-to/add-deps-or-outs-to-a-stage),
+  and `dvc commit` may be needed to finalize the remedy (see link).
+
+- It's also possible to execute stage commands by hand (without `dvc repro`), or
+  to manually modify their output files or directories. Use `dvc commit` to
+  register the changes with DVC once you're done.
+
+  > Note that `dvc unprotect` (or removing the outputs) is usually required
+  > before rewriting files/dirs tracked by DVC.
+
+Note that it's best to try avoiding these scenarios, where the cache,
+`dvc.lock`, and `.dvc` files are force-updated. DVC can't guarantee
+reproducibility in those cases.
```

Comment:
> The 2nd half of the desc. goes over the other scenarios where this might be useful.

BTW... If we don't recommended, why mention them? 😅

Reply:
they are advanced, and users should understand the consequences